src/overlays/actors/ovl_Obj_Skateblock/z_obj_skateblock.c

```diff
@@ -28,76 +39,661 @@ const ActorInit Obj_Skateblock_InitVars = {
+        spD4.y = -10.0f;
+        spD4.z = 0.0f;
+
+        func_80A212F0(&spEC, &spD4, phi_s6);
+
+        spEC.x += this->dyna.actor.world.pos.x;
+        spEC.y += this->dyna.actor.world.pos.y;
+        spEC.z += this->dyna.actor.world.pos.z;
+
+        spE0.x = temp_f28 + spEC.x;
+        spE0.y = spEC.y;
+        spE0.z = temp_f30 + spEC.z;
+
+        if (BgCheck_EntityLineTest3(&globalCtx->colCtx, &spEC, &spE0, &spC8, &spB8, 1, 0, 0, 1, &spBC,
+                                    &this->dyna.actor, 0.0f)) {
+            temp_f0 = Math3D_Vec3fDistSq(&spEC, &spC8);
+            if (temp_f0 < phi_f22) {
+                ret = true;
+                arg2->x = (spC8.x - spEC.x) + (300.0f * this->dyna.actor.scale.z * sp100);
+                arg2->y = 0.0f;
+                arg2->z = (spC8.z - spEC.z) + (300.0f * this->dyna.actor.scale.z * temp_f26);
+                phi_f22 = temp_f0;
+            }
+        }
+    }
+    return ret;
+}
+
+s32 func_80A21C30(ObjSkateblock* this, GlobalContext* globalCtx) {
+    Vec3f sp1C;
+
+    if (func_80A21990(this, globalCtx, &sp1C)) {
+        this->dyna.actor.world.pos.x += sp1C.x;
+        this->dyna.actor.world.pos.z += sp1C.z;
+        return true;
+    }
+
+    return false;
+}
+
+void func_80A21C88(ObjSkateblock* this, s32 arg1) {
+    if ((arg1 == 0) || (arg1 == 1)) {
+        this->unk_16C = &this->dyna.actor.world.pos.x;
+    } else {
+        this->unk_16C = &this->dyna.actor.world.pos.z;
+    }
+    this->unk_170 = arg1;
+}
+
+void func_80A21CB4(ObjSkateblock* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    player->stateFlags2 &= ~0x10;
+    this->dyna.pushForce = 0.0f;
+}
+
+void func_80A21CD8(ObjSkateblock* this) {
+    this->dyna.actor.velocity.y += this->dyna.actor.gravity;
+    if (this->dyna.actor.velocity.y < this->dyna.actor.terminalVelocity) {
+        this->dyna.actor.velocity.y = this->dyna.actor.terminalVelocity;
+    }
+    this->dyna.actor.world.pos.y += this->dyna.actor.velocity.y;
+}
+
+void func_80A21D1C(ObjSkateblock* this) {
+    f32 sp4;
+    f32 temp_f14;
+    f32 temp_f16;
+    f32 temp_f6;
+    f32 phi_f2;
+
+    temp_f14 = this->dyna.actor.world.pos.x - this->dyna.actor.home.pos.x;
+    phi_f2 = ((temp_f14 >= 0.0f) ? 0.5f : -0.5f);
+    sp4 = ((s32)(phi_f2 + temp_f14) / 30) * 30;
+    temp_f14 -= sp4;
+
+    if (fabsf(temp_f14) < 3.0f) {
+        this->dyna.actor.world.pos.x = this->dyna.actor.home.pos.x + sp4;
+    }
+
+    temp_f16 = this->dyna.actor.world.pos.z - this->dyna.actor.home.pos.z;
+    phi_f2 = ((temp_f16 >= 0.0f) ? 0.5f : -0.5f);
+    temp_f6 = ((s32)(phi_f2 + temp_f16) / 30) * 30;
+    temp_f16 -= temp_f6;
+
+    if (fabsf(temp_f16) < 3.0f) {
+        this->dyna.actor.world.pos.z = this->dyna.actor.home.pos.z + temp_f6;
+    }
+}
+
+void func_80A21E9C(ObjSkateblock* this, GlobalContext* globalCtx) {
+    DynaPolyActor* temp_v0;
+    s32 temp_s1;
+    ObjSkateblock* phi_s0;
```

Comment:
```suggestion
    ObjSkateblock* phi_s0;
```
This isn't being used, could it be removed?